File: Java/Queue_Stack/Queue.java
package Java.Queue_Stack;

import java.util.ArrayList;

public class Queue {

    public Queue()
    {
        m_arr = new ArrayList<Integer>();
    };

    public void Enqueue(int item)
    {
        m_arr.add(item);
    }

    public int Dequeue()
    {
        int n = m_arr.get(0);
        m_arr.remove(0);
        return n;
    }

    public int Size()
    {
        return m_arr.size();
    }


    private ArrayList<Integer> m_arr;
    
}

File: JS/Queue_Stack/Queue.js
class Queue
{

    constructor()
    {
        this.items = [];
    }
    enqueue(element)
    {
        this.items.push(element);
    }
    dequeue()
    {
        return this.items.shift();
    }
    peek()
    {
        return this.items[0];
    }
    length()
    {
        return this.items.length ;
    }

}


s = new Queue();
s.enqueue(5);
s.enqueue(51);
s.enqueue(55);
s.enqueue(59);

for(i=0;i<4;i++)
{
    console.log(s.dequeue());
}